### DISCUSSION TITLE
FAQs of Notes1: MODULE_HRLDAS_NETCDF_IO:  Problem finding variable 'XLAT' in NetCDF file

### DISCUSSION THREAD

**Discussion by ZeyuF** [NCAR/hrldas#172] [Q&A]
Hi,
When I run as the tutorial Note1_Single_Point, I get the following error:

./hrldas.exe 
wrfinput_flnm: '../HRLDAS_forcing/run/examples/single_point/hrldas_setup_single_point.nc'
 Problems finding global attribute: ISLAKE; setting to -1
 ncid =        65536

NetCDF: Variable not found                                                      

 ***** MODULE_HRLDAS_NETCDF_IO:  Problem finding variable 'XLAT' in NetCDF file.

My nc file is as follows:

 ncdump -h ../HRLDAS_forcing/run/examples/single_point/hrldas_setup_single_point.nc
netcdf hrldas_setup_single_point {
dimensions:
        west_east = 1 ;
        south_north = 1 ;
        soil_layers_stag = 4 ;
        Time = UNLIMITED ; // (0 currently)

// global attributes:
                :description = "minimum annual vegetation cover" ;
                :units = "%" ;
                :GRID_ID = 1 ;
                :ISWATER = 17 ;
                :ISURBAN = 13 ;
                :ISICE = 15 ;
                :MMINLU = "MODIFIED_IGBP_MODIS_NOAH" ;
                :DX = 0.f ;
                :DY = 0.f ;
                :TRUELAT1 = 0.f ;
                :TRUELAT2 = 0.f ;
                :STAND_LON = 0.f ;
                :MAP_PROJ = 0 ;
                :TITLE = "Created from HRLDAS create_point_data" ;
}

**Comment by tslin2** [comment]
Hi,
the setup file requires the location of latitude and longitude variables, 
        float XLAT(Time, south_north, west_east) ;
        float XLONG(Time, south_north, west_east) ;

**Reply by ZeyuF** [reply]
But does it look like there was already a write to latitude and longitude in the original create_point_data.f90 file(from line 131~137)?

 iret = nf90_def_var(ncid,   "XLAT", NF90_FLOAT, (/dim_id_we,dim_id_sn,dim_id_time/), var_id_xlat)
    iret = nf90_put_att(ncid, var_id_xlat, "description", "latitude")
    iret = nf90_put_att(ncid, var_id_xlat, "units", "degrees_north")

  iret = nf90_def_var(ncid,  "XLONG", NF90_FLOAT, (/dim_id_we,dim_id_sn,dim_id_time/), var_id_xlong)
    iret = nf90_put_att(ncid, var_id_xlong, "description", "longitude")
    iret = nf90_put_att(ncid, var_id_xlong, "units", "degrees_east")

**Reply by cenlinhe** [reply]
could you please double check:
1. if the file path for "hrldas_setup_single_point.nc" in your namelist.hrldas is correctly set?
2. if hrldas_setup_single_point.nc file includes the "XLAT" and other variables using ncview or ncdump (i.e., check if the hrldas_setup_single_point.nc file is correctly generated)?

**Reply by ZeyuF** [reply]
Thank you for your prompt reply！

I made sure my path settings were correct, but I checked the hrldas_setup_single_point.nc file and found that "XLAT" was not generated correctly. But I followed the tutorial Notes1 exactly, and I don't know what the problem is.

hrldas/hrldas/HRLDAS_forcing/run/examples/single_point$ ncdump ./hrldas_setup_single_point.nc 
netcdf hrldas_setup_single_point {
dimensions:
        west_east = 1 ;
        south_north = 1 ;
        soil_layers_stag = 4 ;
        Time = UNLIMITED ; // (0 currently)

// global attributes:
                :description = "minimum annual vegetation cover" ;
                :units = "%" ;
                :GRID_ID = 1 ;
                :ISWATER = 17 ;
                :ISURBAN = 13 ;
                :ISICE = 15 ;
                :MMINLU = "MODIFIED_IGBP_MODIS_NOAH" ;
                :DX = 0.f ;
                :DY = 0.f ;
                :TRUELAT1 = 0.f ;
                :TRUELAT2 = 0.f ;
                :STAND_LON = 0.f ;
                :MAP_PROJ = 0 ;
                :TITLE = "Created from HRLDAS create_point_data" ;
}

**Reply by cenlinhe** [reply]
looks like your hrldas_setup_single_point.nc file does not include any variables. Did you see any error message when generating the hrldas_setup_single_point.nc file?

**Reply by ZeyuF** [reply]
I checked the process of generating the hrldas_setup_single_point.nc file and did not receive any error message. 

It seems to be a problem while generating the nc file. Anyway, I will check my whole workflow again. Sincerely thank you!
